.github/workflows/tests.yml

```diff
@@ -41,14 +38,39 @@ on:
       ccc_credentials:
         required: true
 jobs:
+  matrix:
```

Comment:
since this same step is done multiple times, why not outsource to a reusable action similar to [.github/actions/setup/action.yml](https://github.com/gardenlinux/gardenlinux/blob/a1c09a98f00b42a3428c4bafb3556250b677f551/.github/actions/setup/action.yml)

Reply:
It is not the very same step as it is filtered with different flags, e.g. `--build`,  `--publish`.
But good idea. Will look into it.

Reply:
Fixed this by using a reusable workflow `.github/workflows/generate_matrix.yml` and unique job names.